lightning/mydump/loader.go

```diff
@@ -190,48 +213,60 @@ func (s *mdLoaderSetup) setup(dir string) error {
-				return errors.Errorf("invalid database schema file, duplicated item - %s", fileInfo.path)
+			if _, dbExists := s.insertDB(fileInfo.TableName.Schema, fileInfo.FileMeta.Path); dbExists && s.loader.router == nil {
+				return errors.Errorf("invalid database schema file, duplicated item - %s", fileInfo.FileMeta.Path)
 			}
 		}
 
 		// setup table schema
 		for _, fileInfo := range s.tableSchemas {
-			_, dbExists, tableExists := s.insertTable(fileInfo.tableName, fileInfo.path)
+			_, dbExists, tableExists := s.insertTable(fileInfo.TableName, fileInfo.FileMeta.Path)
 			if !dbExists {
-				return errors.Errorf("invalid table schema file, cannot find db - %s", fileInfo.path)
+				return errors.Errorf("invalid table schema file, cannot find db '%s' - %s", fileInfo.TableName.Schema, fileInfo.FileMeta.Path)
 			} else if tableExists && s.loader.router == nil {
-				return errors.Errorf("invalid table schema file, duplicated item - %s", fileInfo.path)
+				return errors.Errorf("invalid table schema file, duplicated item - %s", fileInfo.FileMeta.Path)
 			}
 		}
 	}
 
 	// Sql file for restore data
 	for _, fileInfo := range s.tableDatas {
-		tableMeta, dbExists, tableExists := s.insertTable(fileInfo.tableName, "")
+		tableMeta, dbExists, tableExists := s.insertTable(fileInfo.TableName, "")
 		if !s.loader.noSchema {
 			if !dbExists {
-				return errors.Errorf("invalid data file, miss host db - %s", fileInfo.path)
+				return errors.Errorf("invalid data file, miss host db '%s' - %s", fileInfo.TableName.Schema, fileInfo.FileMeta.Path)
 			} else if !tableExists {
-				return errors.Errorf("invalid data file, miss host table - %s", fileInfo.path)
+				return errors.Errorf("invalid data file, miss host table '%s' - %s", fileInfo.TableName.Name, fileInfo.FileMeta.Path)
 			}
 		}
-		tableMeta.DataFiles = append(tableMeta.DataFiles, fileInfo.path)
-		tableMeta.TotalSize += fileInfo.size
+		tableMeta.DataFiles = append(tableMeta.DataFiles, fileInfo)
+		tableMeta.TotalSize += fileInfo.Size
 	}
 
-	// Put the small table in the front of the slice which can avoid large table
-	// take a long time to import and block small table to release index worker.
 	for _, dbMeta := range s.loader.dbs {
+		// Put the small table in the front of the slice which can avoid large table
+		// take a long time to import and block small table to release index worker.
 		sort.SliceStable(dbMeta.Tables, func(i, j int) bool {
 			return dbMeta.Tables[i].TotalSize < dbMeta.Tables[j].TotalSize
 		})
+
+		// sort each table source files by sort-key
+		for _, tbMeta := range dbMeta.Tables {
+			dataFiles := tbMeta.DataFiles
+			sort.SliceStable(dataFiles, func(i, j int) bool {
+				if dataFiles[i].FileMeta.SortKey != dataFiles[j].FileMeta.SortKey {
+					return dataFiles[i].FileMeta.SortKey < dataFiles[j].FileMeta.SortKey
+				}
+				return dataFiles[i].FileMeta.Path < dataFiles[j].FileMeta.Path
```

Comment:
```suggestion
				return dataFiles[i].FileMeta.SortKey < dataFiles[j].FileMeta.SortKey
```

the result of WalkDir is already sorted by path, comparing Path is redundant when using SliceStable